### PR TITLE
BUG: error checking before mapping of einsum axes.

### LINKED ISSUE #10792: np.einsum unexpected behavior for double trace on tensors
I run into this error being thrown when doing double summation on tensors with legs with different dimensions.
It looks a lot like #5147 which was solved previously.
This is a minimal example (dimensions and array values are unimportant to the problem):
```
a=np.arange(4).reshape(1,1,2,1,2)
np.einsum('bbcdc->d,a)
```
returns
```ValueError: dimensions in operand 0 for collapsing index 'd' don't match (1 != 2)```.

Summing `b` and `c` separately works as do the following examples:
```
a=np.arange(4).reshape(1,1,2,2,1)
np.einsum('bbccd->d,a)

a=np.arange(8).reshape(1,1,2,2,2)
np.eisum('bbcdc->d,a)
```

### PR DESCRIPTION
**Pull request by jaimefrio**
Fixes #10792.

### CODE REVIEW

File: numpy/core/src/multiarray/einsum.c.src [hunk 1]
```diff
@@ -2225,7 +2225,7 @@ get_combined_dims_view(PyArrayObject *op, int iop, char *labels)
         }
         else {
             /* Update the combined axis dimensions and strides */
-            i = idim + combineoffset;
+            int i = icombinemap[idim + combineoffset];
             if (combineoffset < 0 && new_dims[i] != 0 &&
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/einsum.c.src]: Should this `combineoffset` check come first?

Comment (jaimefrio) [reply numpy/core/src/multiarray/einsum.c.src]: Not really. `i` has to be mapped to the value of `icombine` (which is only advanced when a label is seen for the first time) of the first time this label was found, which was in dimension `idim + combineoffset`, and this has to be done regardless of whether this is the first time we see this label or not.

Comment (eric-wieser) [reply numpy/core/src/multiarray/einsum.c.src]: My thinking is that this could index out of range if `combineoffset` is negative

**Review by eric-wieser — APPROVED** [review]
After a second look, LGTM.

Since you seem to understand what's going on here - if you could add some documentation to `icombinemap`, that would be great - but I'll admit it's out of scope for this fix.

**Comment by jaimefrio** [comment]
I agree on the need to document the code, but since it may take a while it may be better not to hold this fix on that. Have created #10801 to remind us, and I will try to start work on that in the next few days.

**Comment by eric-wieser** [comment]
Great, let's put this in then

**Pull request merged**